src/cli/server/server_config.rs

```diff
@@ -1,30 +1,36 @@
 use std::net::{IpAddr, Ipv4Addr, SocketAddr};
 use std::sync::Arc;
 
-use crate::blueprint::{Blueprint, Http};
-use crate::cli::runtime::init;
-use crate::http::AppContext;
+use crate::app_context::AppContext;
+use crate::blueprint::Http;
+use crate::builder::TailcallExecutor;
 
 pub struct ServerConfig {
-    pub blueprint: Blueprint,
-    pub app_ctx: Arc<AppContext>,
+    pub tailcall_executor: TailcallExecutor,
 }
 
 impl ServerConfig {
-    pub fn new(blueprint: Blueprint) -> Self {
-        let server_context = Arc::new(AppContext::new(
-            blueprint.clone(),
-            init(&blueprint.upstream, blueprint.server.script.clone()),
-        ));
-        Self { app_ctx: server_context, blueprint }
+    pub fn new(mut tailcall_executor: TailcallExecutor) -> Self {
+        let blueprint = tailcall_executor.app_ctx.blueprint.clone();
+        let rt = crate::cli::runtime::init(
+            &blueprint.upstream,
+            tailcall_executor.app_ctx.blueprint.server.script.clone(),
+        );
+        let app_ctx = Arc::new(AppContext::new(blueprint, rt));
+        tailcall_executor.app_ctx = app_ctx;
```

Comment:
why are we re-setting app_ctx here?

Reply:
because earlier the app_ctx was initiated with default runtime and it did not had any information like number of cores. But now we need to rebuild the app_ctx with the blueprint